vcxagency-node/src/service/entities/fwa/entity-fwa.js

```diff
@@ -58,9 +59,18 @@ async function assureFwaWalletWasSetUp (serviceIndyWallets, agencyWalletName, ag
  * There should be only 1 instance of Forward Agent. Forward Agent has circular dependency with Router and Resolver.
  * This is design trade off to have simple singleton.
  */
-async function buildForwardAgent (serviceIndyWallets, serviceStorage, agencyWalletName, agencyWalletKey, agencyDid, agencySeed) {
-  const agencyVerkey = await assureFwaWalletWasSetUp(serviceIndyWallets, agencyWalletName, agencyWalletKey, agencyDid, agencySeed)
-  let router, resolver
+async function buildForwardAgent (serviceIndyWallets, serviceStorage, agencyWalletName, agencyWalletKey, agencyDid, agencySeed, waitTime = 1000, attempts = 10) {
+  let router, resolver, agencyVerkey
+
+  for (let attempt = 0; attempt < attempts; attempt++) {
```

Comment:
Why this turned out to be needed? Is it that sometimes posgres is not yet ready, so we repeatedly try to build up FWA (which tries to connect to its wallet)?
Isn't it sufficient to assure that pgsql is ready for connections?

In case there's some reason why it's tricky and this polling is needed, I'd suggest to inspect the exception type and retry only if it's what we expect it to be.